cpp/oneapi/dal/algo/svm/backend/cpu/train_cls_kernel.cpp

```diff
@@ -167,8 +167,8 @@ static result_t call_binary_daal_kernel(const context_cpu& ctx,
 
     auto arr_label = row_accessor<const Float>{ labels }.pull();
     binary_label_t<Float> unique_label;
-    auto arr_new_label = convert_labels(arr_label, { Float(-1.0), Float(1.0) }, unique_label);
-    const auto daal_labels = interop::convert_to_daal_homogen_table(arr_new_label, row_count, 1);
+    auto daal_labels =
+        convert_labels(labels, arr_label, { Float(-1.0), Float(1.0) }, unique_label, row_count);
```

Comment:
Why do you need to pass both `labels` and `arr_label`? You can pass just `labels` and made `arr_label` local inside that function